staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -104,6 +104,19 @@ func NewEventFromRequest(req *http.Request, level auditinternal.Level, attribs a
 	return ev, nil
 }
 
+// UpdateEventAttribs updates authorizer attributes according to attribs
+func UpdateEventAttribs(ev *auditinternal.Event, attribs authorizer.Attributes) {
+	if user := attribs.GetUser(); user != nil {
+		ev.User.Username = user.GetName()
```

Comment:
I'm not sure if I'm following what this is doing, but `ev.User` should be the "real" user, and `ev.ImpersonatedUser` should hold the the impersonated user attributes. What are we missing?

I agree with @liggitt that the audit event should be thought of as "append only". I.e. we shouldn't change information from earlier stages.